boourns/iceboxer
# Support migrating issues to Canny

This is based on @brentvatne's work on https://github.com/boourns/iceboxer/pull/5. Since then, Product Pains has renamed itself to Canny and their API has varied slightly.

This PR uses the new Canny endpoint URL, and is not React Native specific as envvars are used. It also copies the body of the issue (OP only) into Canny.

## Test Plan

1. Get a GitHub API Token.
2. Log in to Canny. Create a post on the desired board. Inspect the network tab on Chrome to grab the cookie for this particular user (`yourCookie`), as well as the board id (`yourCannyBoardId`).
3. Run the following command:
```
bundle install
GITHUB_API_TOKEN=${yourGitHubAPIToken} ICEBOXER_REPOS="${yourOrg/yourRepo}" CANNY_COOKIE="${yourCookie}" CANNY_BOARD_ID=${yourCannyBoardId} CANNY_COMPANY_NAME=${yourCompanyName} rake icebox
```

At this point I verified that the issues were closed, labeled as "Icebox", and copied over to Canny. See https://react-native.canny.io/feature-requests/p/enabling-watchman-on-windows-causes-js-errors for example.
